ui/src/app/pages/workspace/workspace-library.tsx

```diff
@@ -67,55 +64,78 @@ const LibraryTab: React.FunctionComponent<{
     </Clickable>;
   };
 
+class ProfileState {
+  profile: Profile;
+  reload: Function;
+}
+
+class CurrentTab {
+  title: string;
+  icon: string;
+}
+
+class Props {
+  profileState: ProfileState;
+  enablePublishedWorkspaces: boolean;
+}
+
+class State {
+  currentTab: CurrentTab;
+  errorText: string;
+  workspaceList: WorkspacePermissions[];
+  publishedWorkspaces: WorkspacePermissions[];
+  featuredWorkspaces: WorkspacePermissions[];
+  workspacesLoading: boolean;
+}
+
 export const WorkspaceLibrary = withUserProfile()
-(class extends React.Component<
-  {profileState: { profile: Profile, reload: Function } },
-  {currentTab: {title: string, icon: string}, errorText: string,
-    workspaceList: WorkspacePermissions[], featuredWorkspaces: WorkspacePermissions[],
-    workspacesLoading: boolean}> {
+(class extends React.Component<Props, State> {
   constructor(props) {
     super(props);
     this.state = {
       currentTab: libraryTabEnums.FEATURED_WORKSPACES,
       errorText: '',
       featuredWorkspaces: [],
+      publishedWorkspaces: [],
       workspaceList: [],
       workspacesLoading: true
     };
   }
 
+  libraryTabs = (this.props.enablePublishedWorkspaces || environment.enablePublishedWorkspaces)
```

Comment:
Shouldn't this be "const libraryTabs"?

Reply:
Ah – I didn't realize that this is a class member variable, not a local var. My bad!